src/main/java/xbot/common/trajectory/SwerveSimpleTrajectoryLogic.java

```diff
@@ -144,8 +154,17 @@ public void reset(Pose2d currentPose) {
                     new Pose2d(targetPoint.getTranslation2d(), targetPoint.getRotation2d()));
         }
 
-        if (enableConstantVelocity) {
-            keyPoints = getVelocityAdjustedSwervePoints(initialPoint, keyPoints, constantVelocity);
+        switch (mode) {
+            case KinematicsForIndividualPoints -> {
+                keyPoints = getKinematicsAdjustedSwervePoints(initialPoint, keyPoints);
+            }
+            case KinematicsForPointsList -> {
+                keyPoints = getGlobalKinematicsAdjustedSwervePoints(initialPoint, keyPoints, currentPose);
+            }
+            case ConstantVelocity -> {
+                keyPoints = getVelocityAdjustedSwervePoints(initialPoint, keyPoints, constantVelocity);
+            }
+            default -> {}
```

Comment:
⭐ consider adding logic that fails to compile if every case isn't handled explicitly, that way if a new mode is added later someone can't miss updating this crucial logic